app/services/hts_service/data_cleaning.rb

```diff
@@ -0,0 +1,193 @@
+          AND hiv_status.voided = 0
+          AND hiv_status.concept_id = #{concept('HIV STATUS').id}
+          AND hiv_status.value_coded = #{concept('Positive').id}
+        LEFT JOIN encounter art_init ON art_init.patient_id = p.patient_id
+          AND art_init.voided = 0
+          AND art_init.encounter_type = #{encounter_type('ART Enrollment').id}
+        WHERE e.program_id = #{program('HTC PROGRAM').id}
+          AND e.voided = 0
+          AND DATE(e.encounter_datetime) >= DATE('#{@start_date}')
+          AND DATE(e.encounter_datetime) <= DATE('#{@end_date}')
+          AND art_init.encounter_id IS NULL
+        GROUP BY p.patient_id
+      SQL
+    end
+
+    def duplicate_encounter
+      ActiveRecord::Base.connection.select_all <<~SQL
+        SELECT
+          e.patient_id,
+          e.encounter_type,
+          i.identifier,
+          et.name,
+          DATE(e.encounter_datetime) as visit_date,
+          pn.gender,
+          p.given_name,
+          p.family_name,
+          COUNT(*) total
+        FROM encounter e
+        INNER JOIN encounter_type et ON et.encounter_type_id = e.encounter_type AND et.retired = 0
+        INNER JOIN patient_identifier i ON i.patient_id = e.patient_id AND i.identifier_type = #{patient_identifier_type('National id').id}
+        INNER JOIN person_name p ON p.person_id = e.patient_id AND p.voided = 0
+        INNER JOIN person pn ON pn.person_id = e.patient_id 
+          AND pn.voided = 0
+        WHERE e.program_id = #{program('HTC PROGRAM').id} AND e.voided = 0
+        AND DATE(e.encounter_datetime) >= DATE('#{@start_date}') AND DATE(e.encounter_datetime) <= DATE('#{@end_date}')
+        GROUP BY e.patient_id, e.encounter_type, DATE(e.encounter_datetime)
+        HAVING IF (e.encounter_type = #{encounter_type('VITALS').id}, total > 2, total > 1)
+      SQL
+    end
+
+    # check if clients below 14 years old have a partner reception encounter
+    def partner_status
+      ActiveRecord::Base.connection.select_all <<~SQL
+        SELECT
+          p.patient_id,
+          e.encounter_type,
+          i.identifier,
+          DATE(e.encounter_datetime) as visit_date,
+          pe.gender,
+          pn.given_name,
+          pn.family_name,
+          pe.birthdate
+        FROM patient p
+        INNER JOIN person_name pn ON pn.person_id = p.patient_id 
+          AND pn.voided = 0
+        INNER JOIN person pe ON pe.person_id = p.patient_id
+          AND pe.voided = 0
+        INNER JOIN encounter e ON e.patient_id = p.patient_id 
+          AND e.voided = 0
+          AND e.encounter_type = #{encounter_type('TESTING').id}
+        INNER JOIN patient_identifier i ON i.patient_id = p.patient_id AND i.identifier_type = #{patient_identifier_type('National id').id}
+        INNER JOIN encounter pr ON p.patient_id = pr.patient_id
+          AND pr.voided = 0
+          AND pr.encounter_type = #{encounter_type('Partner Reception').id}
```

Comment:
I think you need to check observations which indicate that the client has a partner